src/Controller/ContactController.php

```diff
@@ -39,22 +43,55 @@ class ContactController extends AbstractController
     public const MAX_NAME_LENGTH = 100;
     public const MAX_MESSAGE_LENGTH = 255;
 
-    public function index()
+    public function index($idProperty = null)
     {
         $errors = [];
         $client = [];
         if ($_SERVER['REQUEST_METHOD'] === 'POST') {
             $client = array_map('trim', $_POST);
             $errors =  $this->validateForm($client, $errors);
+            $propertyRef = "";
             if (empty($errors)) {
-                header('Location: /contact/index/');
+                $name = "Je me presente " . $client['firstname'] . " " . $client['lastname'] . "<br/>";
+                if (!empty($idProperty)) {
+                    $propertyRef = "Je suis interessé par un bien, la ref : " . $idProperty . "<br/>";
+                }
+                $mailPart1 = "";
```

Comment:
dans l'ensemble la gestion de mail semble pas mal. Juste cette partie qui ne va pas.
Quand tu envoies un mail, tu créés une vue (qui est ton mail) donc la construction de ce dernier doit se faire dans un fichier Twig. C'est juste qu'au lieu d'afficher cette vue Twig dans le navigateur en faisant un return, tu envoies cette vue Twig à ta méthode html() du mailer.

en gros tu fait un $emailBody = $this->twig->render('contact/email.html.twig', [$variable1, $variable2...])
et $email->...->html($emailBody);